### PR TITLE
WRN-3832: Migrate IncrementSlider unit tests to Testing Library

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated unit tests from Enzyme to Testing Library for IncrementSlider component.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-3832

### CODE REVIEW

File: IncrementSlider/tests/IncrementSlider-specs.js [hunk 1]
```diff
@@ -1,43 +1,35 @@
...
-	}
-};
-
 describe('IncrementSlider Specs', () => {
 	test('should decrement value', () => {
 		const handleChange = jest.fn();
 		const value = 50;
-		const incrementSlider = mount(
+		render(
 			<IncrementSlider
```
Comment (alexandrumorariu) [inline IncrementSlider/tests/IncrementSlider-specs.js]: Place all render methods in a single line.

File: IncrementSlider/tests/IncrementSlider-specs.js [hunk 2]
```diff
@@ -84,14 +77,19 @@ describe('IncrementSlider Specs', () => {
...
-		incrementSlider.setProps({onChange: handleChange, value: value + 1});
+		rerender(
+			<IncrementSlider
+				onChange={handleChange}
+				value={value + 1}
+			/>
+		);
 
 		const expected = 0;
 		const actual = handleChange.mock.calls.length;
```
Comment (alexandrumorariu) [inline IncrementSlider/tests/IncrementSlider-specs.js]: You can replace this with not.toHaveBeenCalled().

File: IncrementSlider/tests/IncrementSlider-specs.js [hunk 3]
```diff
@@ -66,14 +58,15 @@ describe('IncrementSlider Specs', () => {
...
 				value={value}
 			/>
 		);
+		const incrementButton = screen.getAllByRole('button')[1];
 
-		increment(incrementSlider);
+		tap(incrementButton);
 
 		const expected = 1;
 		const actual = handleChange.mock.calls.length;
```
Comment (alexandrumorariu) [inline IncrementSlider/tests/IncrementSlider-specs.js]: you can replace this with .toHaveBeenCalledTimes(expected).

File: IncrementSlider/tests/IncrementSlider-specs.js [hunk 4]
```diff
@@ -100,474 +98,398 @@ describe('IncrementSlider Specs', () => {
...
 		const icon = 'plus';
-		const incrementSlider = mount(
+		render(
 			<IncrementSlider incrementIcon={icon} />
 		);
 
-		const expected = icon;
-		const actual = incrementSlider.find(`.${css.incrementButton} Icon`).prop('children');
+		const expected = '+';
+		const actual = screen.getAllByRole('button')[1].children[1].children[0];
```
Comment (alexandrumorariu) [inline IncrementSlider/tests/IncrementSlider-specs.js]: I believe you can remove the .children[1].children[0] from here and the test will do the same thing,

File: IncrementSlider/tests/IncrementSlider-specs.js [hunk 5]
```diff
@@ -100,474 +98,398 @@ describe('IncrementSlider Specs', () => {
...
 		const icon = 'minus';
-		const incrementSlider = mount(
+		render(
 			<IncrementSlider decrementIcon={icon} />
 		);
 
-		const expected = icon;
-		const actual = incrementSlider.find(`.${css.decrementButton} Icon`).prop('children');
+		const expected = '-';
+		const actual = screen.getAllByRole('button')[0].children[1].children[0];
```
Comment (alexandrumorariu) [inline IncrementSlider/tests/IncrementSlider-specs.js]: Same here.

**Review by alexandrumorariu — COMMENTED** [review]
LGTM.

**Review by alexandrumorariu — APPROVED** [review]

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Pull request merged**